org.eclipse.scout.rt.client/src/main/java/org/eclipse/scout/rt/client/ui/basic/tree/AbstractTreeNode.java

```diff
@@ -1106,66 +996,29 @@ public Object validateValue(ICell cell, Object value) {
     return value;
   }
 
-  /**
-   * The extension delegating to the local methods. This Extension is always at the end of the chain and will not call
-   * any further chain elements.
-   */
-  protected static class LocalTreeNodeExtension<OWNER extends AbstractTreeNode> extends AbstractExtension<OWNER> implements ITreeNodeExtension<OWNER> {
-
-    public LocalTreeNodeExtension(OWNER owner) {
-      super(owner);
-    }
-
-    @Override
-    public void execDecorateCell(TreeNodeDecorateCellChain chain, Cell cell) {
-      getOwner().execDecorateCell(cell);
-    }
-
-    @Override
-    public void execInitTreeNode(TreeNodeInitTreeNodeChain chain) {
-      getOwner().execInitTreeNode();
-    }
-
-    @Override
-    public void execDispose(TreeNodeDisposeChain chain) {
-      getOwner().execDispose();
-    }
-  }
-
-  protected final void interceptDecorateCell(Cell cell) {
-    List<? extends ITreeNodeExtension<? extends AbstractTreeNode>> extensions = getAllExtensions();
-    TreeNodeDecorateCellChain chain = new TreeNodeDecorateCellChain(extensions);
-    chain.execDecorateCell(cell);
-  }
-
-  protected final void interceptInitTreeNode() {
-    List<? extends ITreeNodeExtension<? extends AbstractTreeNode>> extensions = getAllExtensions();
-    TreeNodeInitTreeNodeChain chain = new TreeNodeInitTreeNodeChain(extensions);
-    chain.execInitTreeNode();
+  private void setDisposing(boolean disposing) {
+    m_flags = FLAGS_BIT_HELPER.changeBit(DISPOSING, disposing, m_flags);
   }
 
-  protected final void interceptDispose() {
-    List<? extends ITreeNodeExtension<? extends AbstractTreeNode>> extensions = getAllExtensions();
-    TreeNodeDisposeChain chain = new TreeNodeDisposeChain(extensions);
-    chain.execDispose();
+  @Override
+  public boolean isDisposing() {
+    return FLAGS_BIT_HELPER.isBitSet(DISPOSING, m_flags);
   }
 
   @Override
   public final void dispose() {
     setDisposing(true);
     try {
-      try {
-        disposeInternal();
-      }
-      catch (RuntimeException e) {
-        LOG.warn("Exception while disposing node.", e);
-      }
-      try {
-        interceptDispose();
-      }
-      catch (RuntimeException e) {
-        LOG.warn("Exception while disposing node.", e);
-      }
+      disposeInternal();
```

Comment:
Did you remove the outer try block by intention? If `disposeInternal` throws an Error, the disposing flag is not restored anymore.